File: app/src/main/java/nasser/com/athletes/Activities/DetailActivity.java
package nasser.com.athletes.Activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import nasser.com.athletes.R;
import nasser.com.athletes.Utils;

public class DetailActivity extends AppCompatActivity {


    private String athleteName, athleteImageURL, athleteBrief;
    private TextView tvAthleteName, tvAthleteBrief;
    private ImageView athleteImage;
    private Bundle bundle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        bundle = getIntent().getExtras();

        setupViews();
        fetchingDataFromBundle(bundle);
        bindingDataToViews();

    }


    private void fetchingDataFromBundle(Bundle bundle) {
        if (bundle != null) {
            athleteName = bundle.getString(Utils.NAME_EXTRA);
            athleteImageURL = bundle.getString(Utils.IMAGE_EXTRA);
            athleteBrief = bundle.getString(Utils.BRIEF_EXTRA);
        }
    }

    private void setupViews() {
        tvAthleteName = (TextView) findViewById(R.id.detailName);
        tvAthleteBrief = (TextView) findViewById(R.id.detailBrief);
        athleteImage = (ImageView) findViewById(R.id.detailImage);
    }

    private void bindingDataToViews() {
        tvAthleteName.setText(athleteName);
        tvAthleteBrief.setText(athleteBrief);
        Glide.with(this).load(athleteImageURL).placeholder(R.drawable.default_imag).into(athleteImage);
    }


}

File: app/src/main/java/nasser/com/athletes/Activities/DetailActivity.kt
package nasser.com.athletes.Activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_detail.*
import nasser.com.athletes.Models.AthleteModel
import nasser.com.athletes.R
import nasser.com.athletes.BusinessConstants

class DetailActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        val athele = extractBundleData()
        bindDataToViews(athele)

    }


    private fun extractBundleData(): AthleteModel.Athele {
        val bundle = intent.extras
        val athele = bundle.get(BusinessConstants.ACTIVITY_EXTRA) as AthleteModel.Athele
        return athele
    }

    private fun bindDataToViews(athele: AthleteModel.Athele) {
        Glide.with(this).load(athele.image).placeholder(R.drawable.default_imag).into(detailImage)
        detailName.text = athele.name
        detailBrief.text = athele.brief
    }

}